samples/SocialSample/Startup.cs

```diff
@@ -199,13 +206,10 @@ 127.0.0.1 MsSecSample.localhost.this
                         await context.Response.WriteAsync("<a href=\"?authscheme=" + type.AuthenticationScheme + "\">" + (type.Caption ?? "(suppressed)") + "</a><br>");
                     }
                     await context.Response.WriteAsync("</body></html>");
-                });
-            });
-
-            // Sign-out to remove the user cookie.
-            app.Map("/logout", signoutApp =>
-            {
-                signoutApp.Run(async context =>
+                    return;
+                }
+                
+                if (context.Request.Path.StartsWithSegments(new PathString("/logout")))
                 {
                     await context.Authentication.SignOutAsync(CookieAuthenticationDefaults.AuthenticationScheme);
                     context.Response.ContentType = "text/html";
```

Comment:
I started porting `AspNet.Security.OpenIdConnect.Server` to the new authentication APIs... and I almost had a heart attack: the new `ChallengeAsync/SignInAsync/SignOutAsync` are really fragile.

For instance, returning a simple `new HttpStatusCodeResult(200)` result from MVC after invoking one of these methods may result in either a no-op thing... or in an "headers already sent" exception thrown by the host :scream: